File: DataStructures and Algorithms/binarySearch/BinarySearch.py
##BinarySearch 
#numbers = list(range(1, 129))
#item = 4
#print(numbers)
#print(len(numbers)) 

def binary_search(list, item):
    low  = 0
    high = len(list)-1 

    while low <= high:
        mid = (low + high)
        guess = list[mid]
        if guess == item:
            return mid 
        if guess >= item:
            high = mid - 1
        else : 
            low = mid + 1 
    return None 

my_list = [1,2, 3, 4, 5, 6, 7, 8,9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]


print(binary_search(my_list, 4))

#print binary_search(my_list, -1)

File: DataStructures and Algorithms/binarySearch/BinarySearch.ts

// Binary search ts 

// say we have an array of numbers 
//const numArr: number[] = [1,2,3,4,5,6,7,8,9,10];
//we are trying to get 2
//const value = 1;

function binarySearch(numArr: number[], value: number): number {
  
    let lo = 0;
    let hi = numArr.length - 1;

    while( lo <= hi){
        const mid = Math.floor((lo + hi) / 2);
        if (numArr[mid] == value){
            return mid;
        }else if (numArr[mid] < value){
            return lo = mid + 1
        }else{
           return hi = mid - 1;
        }
    }

 return - 1;
}

let i = 2
console.log(i)

console.log(binarySearch(numArr, value))
